File: javascript_basics/src/tests/failure_example.test.js
require("../testSetup.js");
const { By } = require("selenium-webdriver");
const { getDriver } = require("../globalSetup.js");
const { logger } = require("../utils/logger.js");
const { sleep } = require("../utils/sleep.js");

describe("Failure Example", function () {

  it("should intentionally fail to test screenshot-on-failure", async function () {

    const driver = getDriver();
    logger.info("STARTING Test --- Intentional Failure Example ---");

    //
    // STEP 1 — Navigate to homepage
    //
    const url = "https://the-internet.herokuapp.com/";
    logger.info(`Navigating to ${url}`);
    await driver.get(url);
    // await sleep(1000);


    //
    // STEP 2 — Intentionally locate something that does not exist
    //
    logger.info("Attempting to locate a NON-EXISTENT element to force failure");

    // This selector is INVALID ON PURPOSE
    await driver.findElement(By.id("this-element-does-not-exist"));
    // await sleep(1000);


    //
    // This line will never be reached (failure occurs above)
    //
    logger.info("This log should never appear, the test should fail earlier");

  });

});

File: csharp_basics/tests/failure_example_test.cs
using NUnit.Framework;
using OpenQA.Selenium;
using Allure.NUnit.Attributes;
using csharp_basics.utils;

namespace csharp_basics.tests
{
    [TestFixture]
    public class FailureExampleTest : BaseTest
    {
        [Test]
        [AllureDescription("Intentional failure to validate screenshot and reporting behavior")]
        public void IntentionalFailureExample()
        {
            Logger.Write("STARTING TEST --- Intentional Failure Example ---");

            string url = "https://the-internet.herokuapp.com/";
            driver.Navigate().GoToUrl(url);
            Logger.Write($"Navigated to {url}");

            Logger.Write("Attempting to locate header and assert incorrect text");
            string headerText = driver.FindElement(By.TagName("h1")).Text;
            Logger.Write($"Found header text: '{headerText}'");

            // This assertion is intentionally wrong to trigger a failure
            Assert.That(
                headerText,
                Is.EqualTo("This text does not exist"),
                "Intentional failure triggered"
            );

            Logger.Write("ENDING TEST --- This line should never execute if the assertion fails ---");
        }
    }
}
